platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/network/connectivity/InboundConnectionHandler.java

```diff
@@ -50,83 +48,57 @@
  */
 public class InboundConnectionHandler {
     private static final Logger logger = LogManager.getLogger(InboundConnectionHandler.class);
-
     private final ConnectionTracker connectionTracker;
     private final NodeId selfId;
-    private final AddressBook addressBook;
     private final InterruptableConsumer<Connection> newConnectionConsumer;
     private final SocketConfig socketConfig;
-    private final boolean doVersionCheck;
-    private final SoftwareVersion softwareVersion;
     /** Rate Limited Logger for SocketExceptions */
     private final RateLimitedLogger socketExceptionLogger;
 
     private final PlatformContext platformContext;
+    private final NetworkPeerIdentifier networkPeerIdentifier;
 
     public InboundConnectionHandler(
             @NonNull final PlatformContext platformContext,
             @NonNull final ConnectionTracker connectionTracker,
+            @NonNull final NetworkPeerIdentifier networkPeerIdentifier,
             @NonNull final NodeId selfId,
-            @NonNull final AddressBook addressBook,
             @NonNull final InterruptableConsumer<Connection> newConnectionConsumer,
-            final boolean doVersionCheck,
-            @NonNull final SoftwareVersion softwareVersion,
             @NonNull final Time time) {
         this.platformContext = Objects.requireNonNull(platformContext);
         this.connectionTracker = Objects.requireNonNull(connectionTracker);
         this.selfId = Objects.requireNonNull(selfId);
-        this.addressBook = Objects.requireNonNull(addressBook);
         this.newConnectionConsumer = Objects.requireNonNull(newConnectionConsumer);
-        this.doVersionCheck = doVersionCheck;
-        this.softwareVersion = Objects.requireNonNull(softwareVersion);
         Objects.requireNonNull(time);
         this.socketExceptionLogger = new RateLimitedLogger(logger, time, Duration.ofMinutes(1));
         this.socketConfig = platformContext.getConfiguration().getConfigData(SocketConfig.class);
+        this.networkPeerIdentifier = networkPeerIdentifier;
     }
 
     /**
-     * Authenticate the peer that has just established a new connection and create a {@link Connection}
+     * Identifies the peer that has just established a new connection and create a {@link Connection}
      *
      * @param clientSocket the newly created socket
      */
-    public void handle(final Socket clientSocket) {
-        SerializableDataInputStream dis = null;
-        SerializableDataOutputStream dos = null;
-        NodeId otherId = null;
+    public void handle(final Socket clientSocket, final List<PeerInfo> peerInfoList) {
         long acceptTime = 0;
+        NodeId otherId = null;
         try {
             acceptTime = System.currentTimeMillis();
             clientSocket.setTcpNoDelay(socketConfig.tcpNoDelay());
             clientSocket.setSoTimeout(socketConfig.timeoutSyncClientSocket());
 
+            if (clientSocket instanceof final SSLSocket sslSocket) {
```

Comment:
we should close the socket if we cant identify the peer

Reply:
If we can, we should refactor this so that we don't rely on an `instanceof` check to make this type of decision. Perhaps when we construct this object, we can configure things differently depending on if we expect SSL sockets or not.